File: Queue/Simple Queue/LinkedList-Based/Queue.py
# Establishing Node
class Node:

    def __init__(self):
        self.data = None
        self.Next = None


# Establishing Queue
class Queue:

    # Initialization
    def __init__(self):
        self.head = None
        self.size = 0

    # Check whether the Queue is Empty or not
    def isEmpty(self):
        return self.head == None

    # Displaying The Queue
    def display(self):

        if self.isEmpty():
            print("Queue is Empty!")
            return

        currentNode = self.head

        while (currentNode != None):
            print("{} <- ".format(currentNode.data), end = "")
            currentNode = currentNode.Next

        print()

    # Enqueue
    def Enqueue(self, x):

        newNode = Node()

        newNode.data = x

        currentNode = self.head

        if self.isEmpty():

            newNode.Next = self.head
            self.head = newNode

        else:

            for i in range(self.size - 1):
                currentNode = currentNode.Next

            newNode.Next = currentNode.Next
            currentNode.Next = newNode

        self.size += 1

    # Dequeue
    def Dequeue(self):

        if self.isEmpty():

            print("Queue is Empty!")
            return -1

        else:

            currentNode = self.head

            value = currentNode.data
            self.head = currentNode.Next

        self.size -= 1

        return value

File: Queue/Simple Queue/LinkedList-Based/Queue.java
public class Queue{

    //Establishing Node
    class Node{

        int data;
        Node next;

    }

    private Node head;
    private int size;

    //Initialization
    Queue(){
        head = null;
        size = 0;
    }

    //Check whether the Queue is Empty or not
    public boolean isEmpty(){
        return head == null;
    }

    //Displaying The Queue
    public void display(){

        if(isEmpty()){
            System.out.println("Queue is Empty!");
            return;
        }

        Node currentNode = head;

        while(currentNode != null){
            System.out.print(currentNode.data + " <- ");
            currentNode = currentNode.next;
        }

        System.out.println();

    }

    //Enqueue
    public void Enqueue(int x){

        Node newNode = new Node();

        newNode.data = x;

        Node currentNode = head;

        if(isEmpty()){

            newNode.next = head;
            head = newNode;

        } else {

            for(int i = 0; i < size - 1; i++)
                currentNode = currentNode.next;

            newNode.next = currentNode.next;
            currentNode.next = newNode;

        }

        size++;

    }

    //Dequeue
    public int Dequeue(){

        int value;

        if(isEmpty()){

            System.out.println("Queue is Empty!");
            return -1;

        } else {

            Node currentNode = head;
            value = currentNode.data;

            head = currentNode.next;

        }

        size--;

        return value;

    }

}
